wagtail/query.py

```diff
@@ -179,6 +185,50 @@ def is_specific(self):
             (SpecificIterable, DeferredSpecificIterable),
         )
 
+    def select_related(self, *fields, for_specific_subqueries: bool = False):
+        """
+        Return a new QuerySet instance that will select related objects.
+
+        If fields are specified, they must be ForeignKey fields and only those
+        related objects are included in the selection.
+
+        If select_related(None) is called, clear the list.
+
+        When called with `for_specific_subqueries=True`, the selected fields
+        will be combined
+        """
+
+        if not for_specific_subqueries:
+            return super().select_related(*fields)
```

Comment:
Suggestion: There is some additional validation in the parent `select_related` (and `prefetch_related`) - should this still be called for `for_specific_subqueries`?

Reply:
I can't think of a nice way to do that without making a query to identify the models. I thought it would be good enough for things to fail at this point for now.

Reply:
Maybe something to include in the documentation as a caveat?